### ISSUE TITLE
Relaxed bounds for `DigestSigner`

### ISSUE THREAD

**Issue by agostbiro** [RustCrypto/elliptic-curves#612]
The `signature` crate only [requires](https://docs.rs/signature/latest/src/signature/signer.rs.html#75-91) a `Digest` for `DigestSigner`, but the `DigestSigner` implementation for [`k256::ecdsa::SigningKey`](https://docs.rs/k256/latest/src/k256/ecdsa/sign.html) introduces more complex bounds that make it [difficult](https://github.com/RustCrypto/traits/issues/1028) to pass custom digests to [`try_sign_digest`](https://docs.rs/k256/latest/k256/ecdsa/struct.SigningKey.html#impl-DigestSigner%3CD%2C%20Signature%3CSecp256k1%3E%3E).

Current bounds:

```rust
impl<D> DigestSigner<D, Signature> for SigningKey
where
    D: CoreProxy + Digest + FixedOutput<OutputSize = U32>,
    D::Core: BlockSizeUser
        + BufferKindUser<BufferKind = Eager>
        + Clone
        + Default
        + FixedOutputCore
        + HashMarker
        + OutputSizeUser<OutputSize = D::OutputSize>,
    <D::Core as BlockSizeUser>::BlockSize: IsLess<typenum::U256>,
    Le<<D::Core as BlockSizeUser>::BlockSize, typenum::U256>: NonZero,
```

I've tried and `k256::ecdsa::SigningKey` also compiles with these trait bounds:

```rust
impl<D> DigestSigner<D, Signature> for SigningKey
where
    D: Digest + FixedOutput<OutputSize = U32>,
```

so I guess `D: CoreProxy` is only needed for the `HashMarker` and `U256` bounds and the digest bounds could be potentially simplified.

[Input from @tarcieri](https://github.com/RustCrypto/traits/issues/1028#issuecomment-1165771055):

> The digest function is also used for RFC6979 deterministic nonce derivation, so that was the original impetus for the more complex bounds, although I think the bounds can be simplified. I think the original ones were to support the use of a generic Hmac, but now it's using SimpleHmac internally.
> 
> The bounds for k256::ecdsa::recoverable::Signature can be simplified even further, because it's now hardcoded to Sha256 for RFC6979.
>

**Comment by tarcieri** [comment]
I think we could potentially relax the bounds as a non-breaking change. AFAICT doing so would only permit things to compile which didn't compile before.

@agostbiro care to open a PR? Otherwise I can look at it soon

**Comment by agostbiro** [comment]
Sure, I've removed all the bounds from `DigestSigner` implementations of `k256::ecdsa::SigningKey` in #613 that aren't needed to compile . Hope I understood correctly :)

**Comment by tarcieri** [comment]
Fixed by #613
